storage/store.go

```diff
@@ -767,10 +768,12 @@ func (s *Store) DisableRaftLogQueue(disabled bool) {
 func (s *Store) ForceRaftLogScanAndProcess(t util.Tester) {
 	// Add each range to the queue.
 	s.mu.RLock()
-	for _, r := range s.replicas {
+	replicas := s.replicas
```

Comment:
This just copies a pointer, not the underlying map, so it is still racy to access `replicas` outside the lock. You need to copy the contents of `s.replicas` into a local slice or map before unlocking s.mu. 

Add a comment that this is pattern is necessary for `ForceRaftLogScanAndProcess` but not the other `ForceXScan` methods because the raft log queue calls back into `Store`. (although if you wanted to refactor this into a single `ForceScanQueueAndProcess` that could work for all queues, it would be fine to apply this pattern to all of them).